### DISCUSSION TITLE
Game crashed after switching from Fullscreen to Borderless

### DISCUSSION THREAD

**Discussion by Phippe** [ppy/osu#15984] [Q&A] [answered]
When I selected `Borderless`, the game froze for a second, then immediately crashed. Now, it won't start up again: It goes as far as displaying a black screen, but afterwards it always crashes. I've also tried rerunning the installer, but it led to the same result.

Edit: Some more info about my setup: Windows 10 21H2, GTX 1070TI, Dual Monitor

### Crash happened:
[database.log](https://github.com/ppy/osu/files/7670200/database.log)
### Ran installer:
[updater.log](https://github.com/ppy/osu/files/7670210/updater.log)
### Opened the game again:
[network.log](https://github.com/ppy/osu/files/7670212/network.log)
[performance.log](https://github.com/ppy/osu/files/7670214/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7670215/runtime.log)
### Older logs (probably not necessary):
[performance-update.log](https://github.com/ppy/osu/files/7670223/performance-update.log)
[performance-draw.log](https://github.com/ppy/osu/files/7670224/performance-draw.log)
[performance-gamethread.audio.log](https://github.com/ppy/osu/files/7670219/performance-gamethread.audio.log)
[performance-gamethread.draw.log](https://github.com/ppy/osu/files/7670220/performance-gamethread.draw.log)
[performance-gamethread.input.log](https://github.com/ppy/osu/files/7670221/performance-gamethread.input.log)
[performance-gamethread.update.log](https://github.com/ppy/osu/files/7670222/performance-gamethread.update.log)

**Comment by bdach** [comment]
Unfortunately I don't see anything out of the ordinary in these logs... Are there any errors in the windows event viewer that look related?

**Comment by nahkd123** [comment]
Can you upload runtime log (and maybe others) beside from the database one **after** the crash?

Oh and yes, event viewer errors might also needed.

**Reply by Phippe** [reply]
All the logs are after the crash. `database.log` was last updated when the crash happened, the others are newer (after I had run the installer, and after I had tried to open the game again).

**Comment by Phippe** [comment, marked as answer]
Whenever I start game and it crashes, I get the following error logged: Display driver "nvlddmkm" stopped responding and has successfully recovered (ID 4101). Might have faulty drivers or whatever, but possibly a non-osu related problem.

**Reply by bdach** [reply]
Yeah that sounds firmly like nvidia's problem. A system reboot may potentially help.

If you wish to try going back to fullscreen you could try editing `framework.ini` in your game files and adjusting it so that `WindowMode = Fullscreen` is in there.

**Reply by Phippe** [reply]
Thanks! In the meantime, I realized the error only happens when Shadowplay is active. I can start the game when it is deactivated, but as soon as I turn on Shadowplay, the game crashes. Maybe a future driver fixes this, but I can't be bothered to fix it myself right now ^^

**Reply by bdach** [reply]
Yeah there are known unexplained issues with shadowplay enabled. Disabling is the recommended workaround for now.

**Reply by Phippe** [reply]
FYI, the error no longer occurs when using the new Direct3D renderer. My guess it that the 1px hack (https://github.com/ppy/osu-framework/pull/5623) causes problems when a dual monitor setup is used. Then again, I don't know anything about renderers and it might have been something else altogether. Fact is, the error is gone :)
